service/controller/app/resource/validation/create.go

```diff
@@ -21,6 +21,12 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 	}
 
 	_, err = r.appValidator.ValidateApp(ctx, cr)
+	if validation.IsAppDependencyNotReady(err) {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "dependent configuration is not ready")
```

Comment:
May be useful for debugging:

```suggestion
		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("dependent configuration is not ready: %#q", err))
```